lib/normalizer.c

```diff
@@ -1105,6 +1105,51 @@ grn_nfkc_normalize_is_middle_dot_family(const unsigned char *utf8_char,
   return GRN_FALSE;
 }
 
+grn_inline static const unsigned char *
+grn_nfkc_normalize_unify_to_katakana(const unsigned char *utf8_char,
+                                     unsigned char *unified,
+                                     size_t length)
+{
+  if (length == 3) {
+    if (utf8_char[0] == 0xe3) {
+      if (utf8_char[1] == 0x81 && utf8_char[2] == 0x80) {
+        /* This character code isn't assigned character */
+        return utf8_char; 
+      } else if (utf8_char[1] == 0x81 &&
+                 utf8_char[2] >= 0x81 && utf8_char[2] <= 0x9f) {
+        /* U+3041 HIRAGANA LETTER SMALL A ..
+         * U+305F HIRAGANA LETTER TA */
+        unified[0] = utf8_char[0];
+        unified[1] = utf8_char[1] + 0x01;
+        unified[2] = utf8_char[2] + 0x20;
+      } else if (utf8_char[1] == 0x81 &&
+                 utf8_char[2] >= 0xa0 && utf8_char[2] <= 0xbf) {
+        /* U+3060 HIRAGANA LETTER DA ..
+         * U+305F HIRAGANA LETTER MI */
+        unified[0] = utf8_char[0];
+        unified[1] = utf8_char[1] + 0x02;
+        unified[2] = utf8_char[2] - 0x20;
+      } else if (utf8_char[1] == 0x82 &&
+                 utf8_char[2] >= 0x80 && utf8_char[2] <= 0x9f) {
+        /* U+3041 HIRAGANA LETTER MU ..
+         * U+305F HIRAGANA LETTER YORI */
+        if ( utf8_char[2] >= 0x97 && utf8_char[2] <= 0x9c ) {
+          /* This character code isn't assigned character */
+          return utf8_char;
+        } else if (utf8_char[2] == 0x9f) {
+          /* This character code has not been inputted into this function.
+           * Because this character already decomposes in grn_nfkc100_decompose */
+          return utf8_char;
+        }
+        unified[0] = utf8_char[0];
+        unified[1] = utf8_char[1] + 0x01;
+        unified[2] = utf8_char[2] + 0x20;
+      }
+    }
+  }
+  return unified;
```

Comment:
Other unify functions use `return unified` immedietly and `return utf8_char` as default style.
Could you use the same style?